framework/include/utils/ChainedReal.h

```diff
@@ -0,0 +1,51 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "libmesh/libmesh_common.h"
+#include "metaphysicl/metaphysicl_version.h"
+
+namespace MetaPhysicL
+{
+#if METAPHYSICL_MAJOR_VERSION < 1
+template <typename, typename>
+class DualNumber;
+#else
+#include "metaphysicl/dualnumber_forward.h"
+#endif
+}
+
+using MetaPhysicL::DualNumber;
+
+typedef DualNumber<Real, Real> ChainedReal;
+
+namespace MetaPhysicL
+{
+#define CompareTypes_default_Types(typenames, typename1, typename2, enabletype)                    \
```

Comment:
this would benefit form a brief comment

Reply:
Agreed. I removed the instantiation here as it's actually not needed. And I added a comment in `ChainedADReal.h`.